website/blog/2023-08-14-mongodb-sorting-scalar.md

```diff
@@ -0,0 +1,166 @@
+
+Below table shows the predefined BSON comparison order for each BSON type.
+
+<!-- use newline in column header for appropriate spacing of columns -->
+<!-- markdownlint-disable MD033 -->
+
+| Order of Comparison<br/>(lowest to highest) | BSON Types                               |
+| ------------------------------------------- | ---------------------------------------- |
+| 1                                           | Null                                     |
+| 2                                           | Numbers (Integer, Long, Double, Decimal) |
+| 3                                           | String                                   |
+| 4                                           | Object                                   |
+| 5                                           | Array                                    |
+| 6                                           | BinData                                  |
+| 7                                           | ObjectId                                 |
+| 8                                           | Boolean                                  |
+| 9                                           | Date                                     |
+| 10                                          | Timestamp                                |
+| 11                                          | Regular Expression                       |
+
+## Comparison of different BSON types
+
+For comparing different BSON types, each BSON type has a predefined order of comparison assigned.
+For example, Null BSON type has the lowest order of comparison 1, so Null is less than any other BSON values.
+Boolean BSON type on the other hand has the order of comparison 8, a BSON type such as ObjectId with a lower order of comparison is less than a Boolean value
+and a BSON type such as Timestamp with a higher order of comparison is greater than a Boolean value.
+
+Comparing different BSON types is merely looking up the order of comparison table for each BSON type and comparing the predefined order.
+There is an exception for Array values, we will be discussing them in another blog post.
+
+### Number comparison
+
+Although numbers have different BSON types, namely Integer, Long, Double and Decimal, they are considered the equivalent BSON type for the purpose of comparison.
+That means comparing numbers consider their values but whether they are Integer, Long, Double or Decimal are not relevant.
+For instance, an Integer value 0 is equivalent to Double 0.0 as far as comparison is concerned.
+
+### Null and non-existent field comparison
+
+For the comparison purpose, non-existent field is equivalent to Null.
+This means that Null and non-existent field are equal as far as comparison is concerned.
+
+## Examples showcasing Sorting for scalar values
+
+Suppose there is an `outfits` collection, following query inserts documents.
```

Comment:
In principle, we don't need this collection to exist to be able to insert documents. Rather the opposite, we need this collection to only contain the documents we insert :) as otherwise we don't know how the order will be affected by the documents existed in the collection before.